zio-query/shared/src/main/scala/zio/query/CompletedRequestMap.scala

```diff
@@ -78,4 +78,33 @@ object CompletedRequestMap {
    */
   val empty: CompletedRequestMap =
     new CompletedRequestMap(Map.empty)
+
+  /**
+   * Constructs a completed requests map from an existing Iterable of tuples
+   */
+  def from[E, A, B](it: Iterable[(A, Exit[E, B])])(implicit ev: A <:< Request[E, B]): CompletedRequestMap =
+    new CompletedRequestMap(Map.from(it))
+
+  private[query] def fromOptional[E, A, B](
+    it: Iterable[(A, Exit[E, Option[B]])]
+  )(implicit ev: A <:< Request[E, B]): CompletedRequestMap = {
+    val builder = Map.newBuilder[A, Exit[E, B]]
+    val iter    = it.iterator
+    while (iter.hasNext) {
+      val (key, exit) = iter.next()
+      exit match {
+        case Exit.Failure(e)       => builder += ((key, Exit.failCause(e)))
+        case Exit.Success(Some(a)) => builder += ((key, Exit.succeed(a)))
+        case Exit.Success(None)    => ()
+      }
+    }
+    from(builder.result())
+  }
+
+  // Only used by Scala 2.12 where the `Map.from` method doesn't exist
+  private implicit class EnrichedMapOps[E, A, B](val map: Map.type) extends AnyVal {
+    def from[K, V](it: Iterable[(K, V)]): Map[K, V] =
+      (Map.newBuilder[K, V] ++= it).result
+  }
```

Comment:
Another option here would be to add a dependency on `scala-collection-compat`, but not sure whether you want to bring in a dependency just for this method